{
    "Comment": "A Catch example of the Amazon States Language using an AWS Lambda function",
    "StartAt": "CreateAccount",
    "States": {
        "CreateAccount": {
            "Type": "Task",
            "Resource": "unknown",
            "Catch": [ {
                "ErrorEquals": ["CustomError"],
                "Next": "CustomErrorFallback"
            }, {
                "ErrorEquals": ["States.TaskFailed"],
                "Next": "ReservedTypeFallback"
            }, {
                "ErrorEquals": ["States.ALL"],
                "Next": "CatchAllFallback"
            } ],
            "End": true
        },
        "CustomErrorFallback": {
            "Type": "Pass",
            "Result": "This is a fallback from a custom Lambda function exception",
            "End": true
        },
        "ReservedTypeFallback": {
            "Type": "Pass",
            "Result": "This is a fallback from a reserved error code",
            "End": true
        },
        "CatchAllFallback": {
            "Type": "Pass",
            "Result": "This is a fallback from any error code",
            "End": true
        }
    }
}
